exec/java-exec/src/test/java/org/apache/drill/exec/planner/logical/TestConvertCountToDirectScan.java

```diff
@@ -159,4 +159,30 @@ public void ensureConvertForSeveralColumns() throws Exception {
     }
   }
 
+  @Test
+  public void ensureCorrectCountWithMissingStatistics() throws Exception {
+    dirTestWatcher.copyResourceToRoot(Paths.get("parquet", "wide_string.parquet"));
+
+    test("use dfs.tmp");
+    String tableName = "wide_str_table";
+    try {
+      // table will contain two partitions: one - with null value, second - with non null value
+      test("create table %s partition by (col_str) as select * from dfs.`parquet/wide_string.parquet`", tableName);
+
+      String query = String.format("select count(col_str) as cnt_str, count(*) as cnt_total from %s", tableName);
+
+      // direct scan should not be applied since we don't have statistics
+      testPlanMatchingPatterns(query, new String[]{}, new String[]{"DynamicPojoRecordReader"});
```

Comment:
No need to create an empty array `new String[]{}`, just null may be passed, since this method has a check for nulls.